core/src/test/java/org/infinispan/dataconversion/ProtostreamTranscoderTest.java

```diff
@@ -0,0 +1,128 @@
+      Mockito.when(registry.getGlobalCtx()).thenReturn(ctx);
+      transcoder = new ProtostreamTranscoder(registry, ProtostreamTranscoderTest.class.getClassLoader());
+      supportedMediaTypes = transcoder.getSupportedMediaTypes();
+   }
+
+   private SerializationContext createCtx() {
+      SerializationContext ctx = ProtobufUtil.newSerializationContext();
+      TestContextInitializer.INSTANCE.registerSchema(ctx);
+      TestContextInitializer.INSTANCE.registerMarshallers(ctx);
+      return ctx;
+   }
+
+
+   @Test
+   @Override
+   public void testTranscoderTranscode() throws Exception {
+      Object transcoded = transcoder.transcode(dataSrc, TEXT_PLAIN, APPLICATION_PROTOSTREAM);
+      assertTrue(transcoded instanceof byte[], "Must be byte[]");
+
+      Object transcodedBack = transcoder.transcode(transcoded, APPLICATION_PROTOSTREAM, TEXT_PLAIN);
+
+      // Must be String as byte[] as sent over the wire by hotrod
+      assertTrue(transcodedBack instanceof byte[], "Must be instance of byte[]");
+      assertEquals(dataSrc, new String((byte[]) transcodedBack, TEXT_PLAIN.getCharset().name()), "Must be equal strings");
+   }
+
+   @Test
+   public void testWrappedMessage() throws IOException {
+      UserObject input = new UserObject("value");
+
+      // Produces MessageWrapped and unwrapped payloads
+      byte[] wrapped = (byte[]) transcoder.transcode(input, APPLICATION_OBJECT, APPLICATION_PROTOSTREAM);
+      byte[] unwrapped = (byte[]) transcoder.transcode(input, APPLICATION_OBJECT, UNWRAPPED_PROTOSTREAM);
+
+      assertEquals(input, ProtobufUtil.fromWrappedByteArray(ctx, wrapped));
+      assertEquals(input, ProtobufUtil.fromByteArray(ctx, unwrapped, UserObject.class));
+
+      // Convert from MessageWrapped back to object
+      Object fromWrapped = transcoder.transcode(wrapped, APPLICATION_PROTOSTREAM, APPLICATION_OBJECT);
+      assertEquals(input, fromWrapped);
+
+      // Convert from unwrapped payload back to object, specifying the object type
+      Object fromUnWrappedWithType = transcoder.transcode(unwrapped, UNWRAPPED_PROTOSTREAM, TYPED_OBJECT);
+      assertEquals(input, fromUnWrappedWithType);
+
+      // Should throw exception if trying to convert from unwrapped without passing the type
+      try {
+         transcoder.transcode(unwrapped, UNWRAPPED_PROTOSTREAM, APPLICATION_OBJECT);
+         Assert.fail("should not convert from unwrapped without type");
+      } catch (MarshallingException ignored) {
+      }
+   }
+
+   @AutoProtoSchemaBuilder(
+         includeClasses = {UserObject.class},
+         schemaFileName = "dataconversion.commons.proto",
+         schemaFilePath = "proto/generated",
+         schemaPackageName = "org.infinispan.dataconversion")
+   interface TestContextInitializer extends SerializationContextInitializer {
+      TestContextInitializer INSTANCE = new TestContextInitializerImpl();
+   }
+
+   static class UserObject {
```

Comment:
Can you make use of one of the POJOs in `TestDataSCI`, say `Key` or `Person`. That way we don't need yet another test class and `SerializationContextInitializer` implementation.

When I originally made ProtoStream the default I managed to remove a lot of duplicated code from the testsuite by removing internal test classes like these. So it would be great if we could stick to the already provided ones. This is a trivial Pojo, so it should work correctly unlike `CountMarshallingPojo` :sweat_smile:

Reply:
Done